include/seqan3/range/container/aligned_sequence_adaptor_constant_access.hpp

```diff
@@ -0,0 +1,688 @@
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+//! \cond DEV
+
+/*! \file
+ *  \brief Provides seqn3::aligned_sequence_adaptor_constant_access.
+ *  \ingroup container
+ *  \author Marie Hoffmann <marie.hoffmann AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <initializer_list>
+#include <type_traits>
+#include <vector>
+
+#include <seqan3/alphabet/concept.hpp>
+#include <seqan3/alphabet/gap/gapped.hpp>
+#include <seqan3/range/container/concept.hpp>
+#include <seqan3/range/detail/random_access_iterator.hpp>
+
+#include "sdsl/bit_vectors.hpp"
+#include "sdsl/rank_support_v5.hpp"
+#include "sdsl/select_support_mcl.hpp"
+#include "sdsl/util.hpp"
+
+namespace seqan3 {
+
+/*!\brief Container for an aligned sequence with random access at constant time.
+ * \tparam inner_type {The container type with which aligned sequence structure
+ * can be constructed, e.g. std::vector<gapped<dna4>>. The inner type of the
+ * container must fulfill the seqan3::alphabet_concept.}
+ * \implements partially the seqan3::random_access_sequence_concept
+ * \remark {This class has to be parameterized with the union type
+ * gapped<alphabet_type> packed into a container supporting the random access
+ * operators [] and at(). This way an aligned sequence can be initialized or
+ * assigned to any sequence of alphabet and/or gap symbols.}
+ *
+ * The gap information of an aligned sequence is internally stored in a bit_vector
+ * set to one whenever an aligned position corresponds to a gap. The ungapped
+ * sequence making up the 0s in the bit vector is stored 'compressed' in a
+ * std::vector.
+ * We use rank and a select functions of the sdsl to map from the gapped
+ * sequence space to the ungapped space and vice versa.
+ * \par Example
+ * '---AT--ATC-GT' is represensed by the bit vector '1110011000100' and the
+ * compressed sequence 'ATATCGT'.
+ */
+
+//!\brief Random access operator concept.
+ template <typename type>
+ concept bool random_access_concept = requires (type val)
+ {
+     // member types
+     typename type::reference;
+     // access container
+     { val[0]    } -> typename type::reference;
+     { val.at(0) } -> typename type::reference;
+ };
+
+template <typename inner_type>
+//!\cond
+    //requires random_access_sequence_concept<inner_t> && alphabet_concept<ranges::v3::value_type_t<inner_type>>
+    requires random_access_concept<inner_type> &&
+    alphabet_concept<ranges::v3::value_type_t<inner_type>>
+//!\endcond
+struct aligned_sequence_adaptor_constant_access
+{
+
+private:
+     //!\privatesection
+    using aligned_sequence_t = aligned_sequence_adaptor_constant_access;
```

Comment:
where is this used?

Reply:
about 6 times in the code.

Reply:
Maybe just make the name of the class shorter? 😆 
Like `aligned_sequence_adaptor` or `gapped_sequence_adaptor`?